tests/test_client.py

```diff
@@ -203,18 +202,32 @@ def test_build_request_object_raises_no_host_exception(self):
         with pytest.raises(NoHostsAvailableException):
             client.build_request_object(None, service, None)
 
-    def test_choose_host(self):
+    def test_choose_host_returns_one_of_the_available_hosts(self):
         hosts = ["foo", "bar", "baz"]
         service = mock.Mock()
         service.get_hosts.return_value = hosts
         assert client.choose_host(service) in hosts
 
-    def test_choose_host_raises_exception(self):
+    def test_choose_host_raises_exception_when_no_hosts_available(self):
         service = mock.Mock()
         service.get_hosts.return_value = []
         with pytest.raises(NoHostsAvailableException):
             client.choose_host(service)
 
+    def test_call_when_raw_response_object_requested(self, mock_response, mock_endpoint):
+        service = mock.Mock()
+        service.get_hosts.return_value = ["http://host1.biz"]
+        service.required_headers = {}
+
+        session = mock.Mock()
+        session.send.return_value = mock_response
+
+        response = client.call(
+            service, mock_endpoint, session=session, logger=mock.Mock(), return_raw_response_object=True
+        )
+
+        assert response is mock_response
```

Comment:
We really do mean `is` here, as we want it to be precisely the object returned by `session.send()`.